src/output-json-ike.c

```diff
@@ -0,0 +1,189 @@
+    LogIKEFileCtx *ikelog_ctx;
+    uint32_t count;
+    MemBuffer *buffer;
+} LogIKELogThread;
+
+bool EveIKEAddMetadata(const Flow *f, uint64_t tx_id, JsonBuilder *js)
+{
+    IKEState *state = FlowGetAppState(f);
+    if (state) {
+        IKETransaction *tx = AppLayerParserGetTx(f->proto, ALPROTO_IKE, state, tx_id);
+        if (tx) {
+            return rs_ike_logger_log(state, tx, LOG_IKE_EXTENDED, js);
+        }
+    }
+
+    return false;
+}
+
+static int JsonIKELogger(ThreadVars *tv, void *thread_data, const Packet *p, Flow *f, void *state,
+        void *tx, uint64_t tx_id)
+{
+    LogIKELogThread *thread = thread_data;
+    JsonBuilder *js = CreateEveHeader(p, LOG_DIR_PACKET, "ike", NULL);
+    if (unlikely(js == NULL)) {
+        return TM_ECODE_FAILED;
+    }
+
+    LogIKEFileCtx *ike_ctx = thread->ikelog_ctx;
+    if (!rs_ike_logger_log(state, tx, ike_ctx->flags, js)) {
+        goto error;
+    }
+
+    MemBufferReset(thread->buffer);
+    OutputJsonBuilderBuffer(js, thread->ikelog_ctx->file_ctx, &thread->buffer);
+    jb_free(js);
+
+    return TM_ECODE_OK;
+
+error:
+    jb_free(js);
+    return TM_ECODE_FAILED;
+}
+
+static void OutputIKELogDeInitCtxSub(OutputCtx *output_ctx)
+{
+    LogIKEFileCtx *ikelog_ctx = (LogIKEFileCtx *)output_ctx->data;
+    SCFree(ikelog_ctx);
+    SCFree(output_ctx);
+}
+
+static OutputInitResult OutputIKELogInitSub(ConfNode *conf, OutputCtx *parent_ctx)
+{
+    OutputInitResult result = { NULL, false };
+    OutputJsonCtx *ajt = parent_ctx->data;
+
+    LogIKEFileCtx *ikelog_ctx = SCCalloc(1, sizeof(*ikelog_ctx));
+    if (unlikely(ikelog_ctx == NULL)) {
+        return result;
+    }
+    ikelog_ctx->file_ctx = ajt->file_ctx;
+
+    OutputCtx *output_ctx = SCCalloc(1, sizeof(*output_ctx));
+    if (unlikely(output_ctx == NULL)) {
+        SCFree(ikelog_ctx);
+        return result;
+    }
+
+    ikelog_ctx->flags = LOG_IKE_DEFAULT;
+    const char *extended = ConfNodeLookupChildValue(conf, "extended");
+    if (extended) {
+        if (ConfValIsTrue(extended)) {
+            ikelog_ctx->flags = LOG_IKE_EXTENDED;
+        }
+    }
+
+    output_ctx->data = ikelog_ctx;
+    output_ctx->DeInit = OutputIKELogDeInitCtxSub;
+
+    AppLayerParserRegisterLogger(IPPROTO_UDP, ALPROTO_IKE);
+
+    result.ctx = output_ctx;
+    result.ok = true;
+    return result;
+}
+
+static TmEcode JsonIKELogThreadInit(ThreadVars *t, const void *initdata, void **data)
+{
+    LogIKELogThread *thread = SCCalloc(1, sizeof(*thread));
+    if (unlikely(thread == NULL)) {
+        return TM_ECODE_FAILED;
+    }
+
+    if (initdata == NULL) {
+        SCLogDebug("Error getting context for EveLogIKE.  \"initdata\" is NULL.");
+        SCFree(thread);
+        return TM_ECODE_FAILED;
+    }
+
+    thread->buffer = MemBufferCreateNew(JSON_OUTPUT_BUFFER_SIZE);
+    if (unlikely(thread->buffer == NULL)) {
+        SCFree(thread);
+        return TM_ECODE_FAILED;
+    }
+
+    thread->ikelog_ctx = ((OutputCtx *)initdata)->data;
```

Comment:
You'll need to do this for multi-threaded eve output support (see `output-json-ikev2.c`
```
 thread->file_ctx = LogFileEnsureExists(thread->ikev2log_ctx->file_ctx, t->id);
    if (!thread->file_ctx) {
        goto error_exit;
    }
```